### PR TITLE
Move globals to getters to allow them to be mocked

### LINKED ISSUE #151: Does not work when mocking `window.fetch`
This is very bad for testing.
when using `ky` tools like pretender can not work, because `ky` stores a reference to `fetch` immediatly after its loaded with `const fetch = getGlobal('fetch');`.

This means when a tool like `pretender` overwrites `window.fetch` for testing `ky` does not use the new `window.fetch` but the stored reference to the *real* fetch and so does a real request.

Is there a good reason to not call `getGlobal('fetch')` whenever `fetch` is required? (currently only [here](https://github.com/sindresorhus/ky/blob/master/index.js#L325) and [here](https://github.com/sindresorhus/ky/blob/master/index.js#L328))

I'm willing to do a PR, but only if it will be merged.

### PR DESCRIPTION
**Pull request by luxferresum**
This will fix #151

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -21,18 +21,27 @@ const getGlobal = property => {
...
 };
 
-const document = getGlobal('document');
-const Headers = getGlobal('Headers');
-const Response = getGlobal('Response');
-const ReadableStream = getGlobal('ReadableStream');
-const fetch = getGlobal('fetch');
-const AbortController = getGlobal('AbortController');
-const FormData = getGlobal('FormData');
+const globals = [
```
Comment (szmarczak) [inline index.js]: Use [`Proxy`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy).

Comment (luxferresum) [reply index.js]: Really? Proxy is not available on all platforms and can not be mocked or transpiled.

Comment (sholladay) [reply index.js]: Personally, I would rather call `getGlobal()` inline where we need it, which will have the same effect. It's a little more verbose, but is much simpler to  read and understand.

Comment (szmarczak) [reply index.js]: @luxferresum

image

Really? You're worried about IE?

@sholladay Yup, that would do the same but IMHO `Proxy` is the way to go. Code is much readable. `globals.URL` > `getGlobal('URL')`

Comment (luxferresum) [reply index.js]: Yes!
Many people still need to support IE!

And I don't think it's much harm to explicit list the globals.

File: index.js [hunk 2]
```diff
@@ -21,18 +21,27 @@ const getGlobal = property => {
...
+	'document',
+	'Headers',
+	'Response',
+	'ReadableStream',
+	'fetch',
+	'AbortController',
+	'FormData'
+].reduce((acc, key) => {
+	Object.defineProperty(acc, key, {
```
Comment (sindresorhus) [inline index.js]: I would prefer using `Object.defineProperties` (note the `s`) and just a normal `for-of` loop.

**Comment by sindresorhus** [comment]
Also make sure that tests are passing.

**Comment by sindresorhus** [comment]
@luxferresum Friendly ping :)

File: index.js [hunk 3]
```diff
@@ -1,46 +1,63 @@
...
+const supportsStreams = typeof globals.ReadableStream === 'function';
+const supportsFormData = typeof globals.FormData === 'function';
 
 const deepMerge = (...sources) => {
 	let returnValue = {};
 
 	for (const source of sources) {
 		if (Array.isArray(source)) {
-			if (!(Array.isArray(returnValue))) {
+		  if (!(Array.isArray(returnValue))) {
```
Comment (szmarczak) [inline index.js]:
```suggestion
			if (!(Array.isArray(returnValue))) {
```

Comment (szmarczak) [reply index.js]: image

but I can edit the file directly... hmm...

**Pull request merged**